### PR TITLE
WIP: Updates to discrimination threshold visualizer.

### LINKED ISSUE #84: Discrimination Threshold Visualizer
Very cool package - I'm interested in starting to contribute at some point in the coming months. 

As a post-learning task, I often find it helpful for classification problems to plot various scores (precision/recall/F1/etc.) over all possible discrimination thresholds (similar to a ROC plot), to gain an intuition for optimal thresholds to use when the model is deployed. Something like this (rough example):

image

Where the x-axis is the threshold, and the y-axis is the score. In this example, I might choose 30 as my threshold to maximize F1, let's say.

Of course, this could be generalized and enriched, etc. Thoughts on this as a potential feature?

### PR DESCRIPTION
**Pull request by bbengfort**
Updates the ThreshViz class to DiscriminationThreshold and implements a few more discrimination features such as F1 score, maximizing arguments and annotations. Documentation and tests are also updated to bring this visualizer into prominence and give more explanation.

Fixes #84

### CODE REVIEW

**Comment by bbengfort** [comment]
TODO:

- [ ] finish documentation 
- [ ] include spam dataset in yellowbrick datasets 
- [ ] index the visualizer correctly (README, index.rst, DESCRPTION, etc)

File: docs/api/classifier/index.rst [hunk 1]
```diff
@@ -3,17 +3,14 @@
...
+Classification models attempt to predict a target in a discrete space, that is assign an instance of dependent variables one or more categories. Classification score visualizers display the differences between classes as well as a number of classifier-specific visual evaluations. We currently have implemented the following classifier evaluations:
+
+-  :doc:`classification_report`: A visual classification report that displays precision, recall, and F1 per-class as a heatmap.
+-  :doc:`confusion_matrix`: A heatmap view of the confusion matrix of pairs of classes in multi-class classification.
+-  :doc:`rocauc`: Graphs the receiver operating characteristics and area under the curve.
+-  :doc:`class_balance`: Visual inspection of the target to show the support of each class to the final estimator.
+-  :doc:`class_prediction_error`: An alternative to the confusion matrix that shows both support and the difference between actual and predicted classes.
+-  :doc:`threshold`: Shows precision, recall, f1, and queue rate over all thresholds for binary classifiers that use a discrimination probability or score.
```
Comment (ndanielsen) [inline docs/api/classifier/index.rst]: show this name be updated and corresponding documentation name?

File: docs/api/classifier/threshold.rst [hunk 2]
```diff
@@ -1,45 +1,58 @@
 .. -*- mode: rst -*-
```
Comment (ndanielsen) [inline docs/api/classifier/threshold.rst]: maybe we should update this document name?

File: docs/api/classifier/index.rst (hunk 1 above)
Comment (ndanielsen) [reply docs/api/classifier/index.rst]: nevermind

File: docs/api/classifier/threshold.rst (hunk 2 above)
Comment (ndanielsen) [reply docs/api/classifier/threshold.rst]: nevermind it's clear

**Pull request merged**